### DISCUSSION TITLE
Changing screen resolution

### DISCUSSION THREAD

**Discussion by woomia** [helloSystem/hello#241] [General]
I love, helloSystem! But can you change the screen resolution when running from the ISO? I'm using 0.6 on Intel® HD Graphics 4400 (HSW GT2).

Thanks

**Comment by probonopd** [comment]
Yes, System -> Preferences -> Screen Settings -> right-click on your screen.

@PreyK was working on a [native tool](https://github.com/helloSystem/Utilities/issues/41#issuecomment-869164337).

**Reply by guckux** [reply]
Uh, which fast support! ;)
I am coming from FreeBSD (native), trying out different "forks" for desktop-ready installations...

At the moment, I've tried to install hello 0.6 on my Thinkpad W520 with i7 2860QM and Nvidia Quadro 2000M (configured in discrete mode in BIOS). No Nvidia or intel frame buffer was loaded - screen resolution is 640x480 - so the GUI-installer was shown, but the continue button is out of the visible screen.
Also, with your hint I've look for changing the resolution, but no other resolution is offered...

Thanks!

**Reply by guckux** [reply]
Same problem - only resolution of 640x480 possible, if I deactivate the Nvidia graphics card using the intel frame buffer...

**Reply by guckux** [reply]
Resolved: The systems installs Nvidia version 440 drivers, which didn't support the Nvidia Quadro 2000, after installing with pkg the Nvidia 390 driver version (following the installation instructions) it works fine.

Thanks for the maintainers

**Reply by probonopd** [reply]
Thanks @guckux. I was hoping we would get around with just the latest version of the Nvidia drivers on the ISO, but it seems like we need to ship all of them, which is a pity.

https://github.com/helloSystem/ISO/blob/74b5ccf3ea17ffef522a6557162b1724751e27a3/build.sh#L338-L339

**Reply by probonopd** [reply]
Added the 390 driver. Should be in the next experimental build.

**Reply by guckux** [reply]
Thanks @probonopd

**Reply by probonopd** [reply]
Are you seeing any change when running https://github.com/helloSystem/ISO/releases/tag/experimental-13.0 @guckux?

**Reply by guckux** [reply]
@probonopd 

Great work, did 2 short tests:
1) Booting with HD3000 only enabled, screen switching to full resolution, seems so far so fine.
2) Booting with Nvidia in Optimus mode with "System detect Optimus" - seems to be fallback to HD3000... for me, doesn't matter... Booting with dedicated Nvidia (discrete graphics) no Optimus, booting from stick with full resolution (boot messages), X11 has detected the 1920x1080 resolution, installing.
Rebooting the newly installed system, the boot messages are in VGA resolution (doesn't matter for me, cosmetics :D ), X11 starts with supported Nvidia 390 version - GREAT! WoW!
Good work done for FreeBSD 13-RELEASE dsitribution ;) .

**Reply by probonopd** [reply]
> Optimus

Since I have no Optimus hardware, I cannot test that is needed for it. But there are some packages:

```
nvidia-hybrid-graphics-0.5     NVIDIA secondary GPU configuration - Optimus Technology support
nvidia-hybrid-graphics-390-0.5 NVIDIA secondary GPU configuration - Optimus Technology support
nvidia-secondary-driver-470.74_1 NVidia graphics card binary drivers for hardware OpenGL rendering on secondary device
nvidia-secondary-driver-390-390.144 NVidia graphics card binary drivers for hardware OpenGL rendering on secondary device
nvidia-settings-470.74         Display Control Panel for X NVidia driver
nvidia_gpu_prometheus_exporter-g20181028_1 NVIDIA GPU Prometheus exporter
```

If you can tell me what needs to be installed and how it needs to be configured, then maybe we can make it work?

> boot messages are in VGA resolution

Yes, right now I am focusing on the Live image. Do you get the native resolution on that machine when booting the Live image? If yes, then it should be easy to apply the same setting to the installed version. Still working on that one.

**Reply by guckux** [reply]
Yes, booting from Stick in Live image supports full the high resolution.

Optimus depends - so far as I know - only to Laptops. I recognized first time with the actual helloSystem that they are available... Sorry for my nescience for this case - but I will have one look more for you!
